app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js

```diff
@@ -47,6 +52,25 @@ function findExistingNetwork(chainId, findNetworkConfigurationBy) {
   return findNetworkConfigurationBy({ chainId });
 }
 
+export function findExistingNetworkByNetworkClientId(
```

Comment:
is there a reason this needs to be different when its an infura network

Reply:
I think we can get rid of this now actually - at some point I was trying to make the switch ethereum chain dialog show the selectedNetwork for the dapp instead of the metamask globaally selected network. 

Mark reviewed it with me and it was deemed best if it just remains as it was for now, and we can improve / iterate on the switchEthereumChain dialog afterwards

Reply:
okay its gone!